doc/releases/changelog-dev.md

```diff
@@ -4,6 +4,57 @@
 
 <h3>New features since last release</h3>
 
+* PennyLane now has a new utility class ``qml.ArgMap``.
+  [(#1645)](https://github.com/PennyLaneAI/pennylane/pull/1645)
+
+  It acts as a dictionary with a restricted type of keys in the context of a given ``QNode``.
+  Each key takes the form ``tuple[int, tuple[int] or int]``. The first ``int`` in the
+  tuple describes the argument index of the ``QNode``, while the second entry describes the
+  index within that argument. That is ``(2, (0, 3))`` describes the parameter in the
+  ``0``th row and ``3``rd column of the ``2``nd argument of a QNode. The second entry of the
+  tuple may be an ``int`` instead, indexing a one-dimensional array.
+
+  The main feature of ``ArgMap`` is its interpretation of keys handed to it when
+  reading or writing entries: If they are uniquely describing a parameter, incomplete
+  keys may be passed to the ``ArgMap``.
+
+  For example, the usage as dictionary works as usual:
+
+  ```pycon
+  >>> argmap = qml.ArgMap({
+  ...     (0, 0): "First parameter in first argument",
+  ...     (0, 1): "Second parameter in first argument",
+  ...     (1, None): "Scalar second argument",
+  ...     (2, (0, 4, 2)): "Selected parameter in 3D array in third argument",
+  ... })
+  >>> argmap[(0, 0)]
+  'First parameter in first argument'
+  ```
+
+  However, as the second argument is a scalar, it is sufficient to pass the corresponding single
+  integer to the ``ArgMap``:
+
+  ```pycon
+  >>> argmap[1]
+  'Scalar second argument'
+  ```
+
+  Similarly, for a single-argument ``ArgMap``, we get short-hand access to the entries of the
+  only argument:
+  ```pycon
+  >>> argmap = qml.ArgMap(
+  ...     {
+  ...          (None, 0): "First parameter in only argument",
+  ...          (None, 1): "Second parameter in only argument",
```

Comment:
Would `(0, 0)` and `(0, 1)` also work here?

Reply:
(Note however, that `(None, 0)` and `(None, 1)` are no longer valid keys, then!)